src/main/java/graphql/execution/directives/QueryDirectives.java

```diff
@@ -32,6 +33,13 @@
      */
     Map<String, List<GraphQLDirective>> getImmediateDirectivesByName();
 
+    /**
+     * This will return a map of the applied directives that are immediately on a merged field
+     *
+     * @return a map of all the applied directives immediately on this merged field
+     */
+    Map<String, List<GraphQLAppliedDirective>> getImmediateAppliedDirectivesByName();
```

Comment:
I think the bigger question is actually if applied directives should be used for queries AND schemas? conceptually they are the same, but code wise `GraphQLApplieDirective` is a schema element and not a query element